src/setfit/onnx.py

```diff
@@ -0,0 +1,227 @@
+from .modeling import SetFitModel
+from sentence_transformers import models
+import numpy as np
+import onnx
+import warnings
+from typing import Optional, Union
+import torch
+from sklearn.linear_model import LogisticRegression
+
+
+class ONNXSetFitModel(torch.nn.Module):
+    def __init__(
+        self,
+        model_body,
+        pooler,
+        model_head: Optional[Union[torch.nn.Module, LogisticRegression]] = None,
+    ):
+        super().__init__()
+
+        self.model_body = model_body
+        self.pooler = pooler
+        self.model_head = model_head
+
+    def forward(self, input_ids, attention_mask, token_type_ids):
+        hidden = self.model_body(input_ids, attention_mask, token_type_ids)
+        hidden = {"token_embeddings": hidden[0], "attention_mask": attention_mask}
+
+        hidden = self.pooler(hidden)["sentence_embedding"]
```

Comment:
nit:
```suggestion
        embeddings = self.pooler(hidden)["sentence_embedding"]
```

Also is the `sentence_embedding` key necessary? I was thinking we could just require people to return a `torch.Tensor`

Reply:
Great suggestion! Yeah if we remove this key access then people can use other poolers more easily.

Reply:
I think `sentence_embedding` is necessary since now we use `sentence-transformers`'s pooling layer.
Maybe instead of remove it, adding a if-else to check whether the pooler comes from `sentence-transformers`?

Reply:
I was thinking to make the type of Pooler `Optional[Union[nn.Module, Callable[[torch.Tensor], torch.Tensor]]]` Then to pass in the Pooler like we did before you could do:

```python
pooling_layer = model_body._modules["1"]
model_pooler = lambda x: pooling_layer(x)["sentence_embedding"]
OnnxSetFitModel(transformer, model_pooler, model_head).cpu()
```

This would let people pass in normal Callable's too. Thoughts?